### PR TITLE
fix(sdk-core): fix lightning requests params

### PR DESCRIPTION
**Pull request by sj3014**
- GetInvoices and CreateDepositAddress do not take params
- add required params for LightningWithdrawal
- add optional params for PayInvoice
- add maximumBalance to response for GetBalance

Ticket: BG-54079

### CODE REVIEW

**Comment by tylerlevine** [comment]
Follow up for post merge comments on https://github.com/BitGo/BitGoJS/pull/2613

File: modules/sdk-core/src/bitgo/lightning/iLightning.ts [hunk 1]
```diff
@@ -155,10 +149,11 @@ export type GetInvoicesResponse = t.TypeOf<typeof GetInvoicesResponse>;
...
 export interface ILightning {
   createInvoice(params?: CreateInvoiceParams): Promise<CreateInvoiceResponse>;
-  createDepositAddress(params?: CreateDepositAddressParams): Promise<CreateDepositAddressResponse>;
+  createDepositAddress(): Promise<CreateDepositAddressResponse>;
   payInvoice(params?: PayInvoiceParams): Promise<PayInvoiceResponse>;
   getBalance(): Promise<GetBalanceResponse>;
   withdraw(params?: LightningWithdrawalParams): Promise<WithdrawResponse>;
   deposit(params?: LightningDepositParams): Promise<DepositResponse>;
-  getInvoices(params?: GetInvoicesParams): Promise<GetInvoicesResponse>;
+  // Implemented in lnd-middleware but not in wallet-platform
```
Comment (tylerlevine) [inline modules/sdk-core/src/bitgo/lightning/iLightning.ts:157]: This comment isn't helpful for those who don't have context on the lightning implementation - please remove.

File: modules/sdk-core/src/bitgo/lightning/iLightning.ts [hunk 2]
```diff
@@ -155,10 +149,11 @@ export type GetInvoicesResponse = t.TypeOf<typeof GetInvoicesResponse>;
...
+  getInvoices(): Promise<GetInvoicesResponse>;
```
Comment (tylerlevine) [inline modules/sdk-core/src/bitgo/lightning/iLightning.ts:158]: I think we should just remove this function from the interface - it needs to take some params and we don't know what they are yet. We can add it later once we added the necessary backend support.

File: modules/sdk-core/src/bitgo/lightning/iLightning.ts [hunk 3]
```diff
@@ -6,21 +6,10 @@ export interface CreateInvoiceParams {
...
-  id?: string;
-  status?: 'open' | 'closed';
-}
-
 export interface LightningWithdrawalParams {
   amount: number;
-  address?: string;
-  sequenceId?: string;
+  address: string;
+  sequenceId: string;
```
Comment (tylerlevine) [inline modules/sdk-core/src/bitgo/lightning/iLightning.ts:12]: These should be optional

File: modules/sdk-core/src/bitgo/lightning/lightning.ts [hunk 4]
```diff
@@ -49,7 +47,7 @@ export class Lightning implements ILightning {
     throw new Error('method not implemented');
   }
 
-  public async getInvoices(params?: GetInvoicesParams): Promise<GetInvoicesResponse> {
+  public async getInvoices(): Promise<GetInvoicesResponse> {
```
Comment (tylerlevine) [inline modules/sdk-core/src/bitgo/lightning/lightning.ts:50]: Let's remove until we're ready to implement this and we know what the params should be.

**Review by tylerlevine — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/lightning/iLightning.ts (hunk 2 above)
Comment (sj3014) [reply modules/sdk-core/src/bitgo/lightning/iLightning.ts:158]: Should we also remove GetInvoicesParams and GetInvoicesResponse types?

Comment (tylerlevine) [reply modules/sdk-core/src/bitgo/lightning/iLightning.ts]: yeah, let's remove all the code related to `getInvoices()` and we'll add it all back when we're ready.

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**